spec/book_spec.rb

```diff
@@ -0,0 +1,45 @@
+require_relative '../book'
+require_relative '../rental'
+require_relative '../person'
+
+RSpec.describe Book do
+  let(:title) { 'Harry Potter' }
+  let(:author) { 'J. K. Rowling' }
+  let(:book) { Book.new(title, author) }
+
+  describe '#initialize' do
+    it 'creates a new book with the correct title and author' do
+      expect(book.title).to eq(title)
+      expect(book.author).to eq(author)
+    end
+
+    it 'creates a new book with an empty rentals array' do
+      expect(book.rentals).to be_an(Array)
+      expect(book.rentals).to be_empty
+    end
+  end
+
+  describe '#add_rental' do
+    it 'creates a new rental for the book and person' do
+      person = Person.new(28, 'Biftu Girma')
+      date = '2023/08/03'
+      rental = person.add_rental(book, date)
+
+      expect(rental).to be_a(Rental)
+      expect(rental.book).to eq(book)
+      expect(rental.person).to eq(person)
+      expect(rental.date).to eq(date)
+    end
+
+    it "adds the rental to the book's rentals array" do
+      person = Person.new(24, 'Peter A.')
+      date = '2023/08/03'
+      person.add_rental(book, date)
+
+      expect(book.rentals.length).to eq(1)
+      expect(book.rentals.first).to be_a(Rental)
+      expect(book.rentals.first.person).to eq(person)
+      expect(book.rentals.first.date).to eq(date)
+    end
+  end
+end
```

Comment:
- [ ] As mentioned by the previous reviewer and it is required for the project  to add tests to all classes that you created while implementing the Decorator design pattern(e.g. : `BaseDecorator`, `CapitalizeDecorator`, `TrimmerDecorator`, `Nameable` ), kindly ensure that you also create specific spec file to each one with tests and add test to each method inside.